src/pages/settings/Profile/PersonalDetails/AddressPage.js

```diff
@@ -52,179 +52,158 @@ const defaultProps = {
     },
 };
 
-class AddressPage extends Component {
-    constructor(props) {
-        super(props);
-
-        this.validate = this.validate.bind(this);
-        this.updateAddress = this.updateAddress.bind(this);
-        this.onCountryUpdate = this.onCountryUpdate.bind(this);
-
-        const currentCountry = lodashGet(this.props.privatePersonalDetails, 'address.country') || '';
-        const currentCountryISO = PersonalDetails.getCountryISO(currentCountry) || CONST.COUNTRY.US;
-        const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [currentCountryISO, 'samples'], '');
-        this.state = {
-            isUsaForm: currentCountry === CONST.COUNTRY.US || currentCountry === CONST.USA_COUNTRY_NAME,
-            zipFormat: this.props.translate('common.zipCodeExampleFormat', {zipSampleFormat}),
-        };
-    }
+/**
+ * @param {Function} translate - translate function
+ * @param {Boolean} isUsaForm - selected country ISO code is US
+ * @param {Object} values - form input values
+ * @returns {Object} - An object containing the errors for each inputID
+ */
+const validate = (translate, isUsaForm, values) => {
```

Comment:
Why would we prefer that? sounds like more complexity to me.

Reply:
Is it to avoid redefining the callback `(values) => validate(props.translate, isUsaForm, values)` each renderer? that would make sense to me.

Let me know if this should be changed.

Reply:
Yes, I think consistency is important, I'll change it. Also, looking at it again, I'm passing down an new instance of a function every render, and this can make it hard later for other components to use `useCallback`/`useMemo` if they watch this callback.

I think it is better to use `useCallback` as you mentioned. The only exception for me would be if the function doesn't depend at all on props, then I think it is better defined outside the component.

Reply:
These are the [guidelines](https://github.com/Expensify/App/blob/4b1c3264cb0d94c7eba5c89b86003934b0ecd3ac/contributingGuides/STYLE.md#should-i-wrap-all-my-inline-functions-with-usecallback-or-move-them-out-of-the-component-if-they-have-no-dependencies), so either way works fine.. Although in this case we are passing a new lambda to Form, so it might be better to create a stable version of the function with a useCallback inside the component,

Reply:
Do you know that happens with the function comment when we put it inside the component, inside a `useCallback`? do we keep using comment block `./* */`? or `//`? do we still describe each param/return?
<img width="561" alt="image" src="https://github.com/Expensify/App/assets/87341702/029dbf2d-afbd-4aaa-a08b-5471b12a2fc4">

I feel like I would get rid of a documentation comment like this because we already describe what the props being pass to the component are.

Reply:
It seems like we keep [docs](https://github.com/Expensify/App/blob/dcb6c2ca9950602c48e869380db120d3c0b79099/src/pages/home/report/ReportActionsList.js#L111-L116) around.

Reply:
Brought back the documentation comment.